### DISCUSSION TITLE
`qb.insert()` is not `returning` all columns

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2551] [Q&A] [answered]
`await qb.insert(dto).withSchema(schema)` does not return (via `returning`) all columns, namely it does not return two timestamp columns. I also checked the generated query (using `qb.insert(dto).withSchema(schema).getQuery()`, but it confirmed this problem.

Is it possible to configure `insert` queries to return all columns (via `returning *`? Or is it possible to specify which column to return?

**Comment by B4nan** [comment, marked as answer]
Only columns with database defaults are returned. You probably do not define the defaults in your entity definition?

**Reply by tukusejssirs** [reply]
I do, but only where I need the defaults. I need to revalute this decision.

Anyway, is `[o]nly columns with database defaults are returned` noted somewhere in the docs? If not, I’d suggest to do so. I’ve search the docs for `returning`, but I found nothing related. :man_shrugging:

Therefore I cannot override this rule?

**Reply by tukusejssirs** [reply]
Actually, I did set the defaults for all columns in this entity, except for the two timestamp columns, where I set the defaults via value (e.g. `property = new Date()`).

**Reply by tukusejssirs** [reply]
Adding `default: Date.now()` fixed the issue, although the [docs](https://mikro-orm.io/docs/next/defining-entities/#default-values) for `ts-morph` (which I use) state that I can define the default value like `baz = new Date()`, therefore I _presume_ it should work, but it does not.

**Reply by B4nan** [reply]
Runtime default is not database default. I dont understand why you expect any column with a runtime default to use returning statement. From the ORM point of view, the property has no db default, hence no reason to reload the value after insertion. The fact that you have such db default in your schema is irrelevant, it needs to be in metadata, as that is what the ORM sees. 

This has nothing to do with metadata providers.

**Reply by tukusejssirs** [reply]
This is my first time with ORM, so bare with me please. :pray: Now I have even better understanding how MikroORM works.
